### ISSUE TITLE
Deadlock running integration test ProfileReportBackwardsCompatTest.sequentialSingleThreadJTPTest()

### ISSUE THREAD

**Issue by CoreRasurae** [Syncleus/aparapi#101]
It looks like some of the threads die during kernel execution, at least there should be 16 threads and I only have 7 running and they are all waiting on a CyclicBarrier on KernelRunner.await() line 331 called from KernelRunner.access() line 679 called from KernelRunner.run() line 663. Could it be some exception that is thrown that makes the threads die and then there aren't enough parties for the CyclicBarrier to unlock?
It happened running sequentialSingleThreadJTPTest().

**Comment by CoreRasurae** [comment]
I am able to reproduce the bug before the feature was merged at commit cd911b937c7804dac495f5fadd573fe1f2ac5a4f -  test: added another test showing out of scope error. 

With the attached test - 
[ProfileReportBackwardsCompatTest.java.txt](https://github.com/Syncleus/aparapi/files/1935359/ProfileReportBackwardsCompatTest.java.txt)

**Comment by CoreRasurae** [comment]
I was also able to reproduce the bug with aparapi releases 1.4.1 and 1.6.0.

**Comment by freemo** [comment]
@CoreRasurae Thanks.. I am worried this might be a tricky one to catch, particularly since it has yet to occur for me locally, only on travis. But ill keep poking around and see what i come up with.

**Comment by automenta** [comment]
pitest might be useful here to discover test deficiencies http://pitest.org/

**Comment by freemo** [comment]
@automenta i got them to use pitest at comcast. Its slow but can be useful

**Comment by CoreRasurae** [comment]
@freemo I think the easiest way to reproduce the problem is to run it on a dual core machine, where the number of cores are restricted.

**Comment by freemo** [comment]
This bug has been resolved, closing.
